test/apps/iodemo/io_demo.cc

```diff
@@ -685,8 +685,7 @@ class DemoClient : public P2pDemoCommon {
     }
 
     inline void check_time_limit(double current_time) {
-        if ((_status == OK) &&
```

Comment:
so there was case when _status was not OK during check_time_limit() ?
why did it hang?
maybe caller of check_time_limit() should check _status!=OK?

Reply:
> so there was case when _status was not OK during check_time_limit() ?

yes, error handling callback was called and `_status = ERROR`

> why did it hang?

so we don't stop by timeout

> maybe caller of check_time_limit() should check _status!=OK?

I don't think so, since we should stop the execution if run-time exceeded on a client

Reply:
seems we need to set `_status = OK` before update_retry() instead of run() - on the client

Reply:
done